### ISSUE TITLE
Validation errors and crash in Buffer device address, Descriptor indexing, Timeline semaphore

### ISSUE THREAD

**Issue by danilw** [KhronosGroup/Vulkan-Samples#377]
**Linux Nvidia driver 470.74**

On resize every of listed Samples Window become black(no image) and application not respond (freeze/crash).

**Buffer device address** `vulkan_samples sample buffer_device_address` - launch with no crash, but two Validation errors:
```
VUID-VkShaderModuleCreateInfo-pCode-04147(ERROR / SPEC): msgNum: 1028204675 - Validation Error: [ VUID-VkShaderModuleCreateInfo-pCode-04147 ] Object 0: handle = 0x353ec08, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x3d492883 | vkCreateShaderModule(): The SPIR-V Extension (SPV_EXT_physical_storage_buffer) was declared, but none of the requirements were met to use it. The Vulkan spec states: If pCode declares any of the SPIR-V extensions listed in the SPIR-V Environment appendix, one of the corresponding requirements must be satisfied (https://vulkan.lunarg.com/doc/view/1.2.170.0/linux/1.2-extensions/vkspec.html#VUID-VkShaderModuleCreateInfo-pCode-04147)
    Objects: 1
        [0] 0x353ec08, type: 3, name: NULL
VUID-VkShaderModuleCreateInfo-pCode-04147(ERROR / SPEC): msgNum: 1028204675 - Validation Error: [ VUID-VkShaderModuleCreateInfo-pCode-04147 ] Object 0: handle = 0x353ec08, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x3d492883 | vkCreateShaderModule(): The SPIR-V Extension (SPV_EXT_physical_storage_buffer) was declared, but none of the requirements were met to use it. The Vulkan spec states: If pCode declares any of the SPIR-V extensions listed in the SPIR-V Environment appendix, one of the corresponding requirements must be satisfied (https://vulkan.lunarg.com/doc/view/1.2.170.0/linux/1.2-extensions/vkspec.html#VUID-VkShaderModuleCreateInfo-pCode-04147)
    Objects: 1
        [0] 0x353ec08, type: 3, name: NULL
```

**Crash on resize**:
```
VUID-vkQueueSubmit-pWaitSemaphores-00069(ERROR / SPEC): msgNum: -276461468 - Validation Error: [ VUID-vkQueueSubmit-pWaitSemaphores-00069 ] Object 0: handle = 0x130000000013, type = VK_OBJECT_TYPE_SEMAPHORE; Object 1: handle = 0x3405438, type = VK_OBJECT_TYPE_QUEUE; | MessageID = 0xef858864 | vkQueueSubmit(): pSubmits[0].pWaitSemaphores[0] Queue VkQueue 0x3405438[] is waiting on semaphore (VkSemaphore 0x130000000013[]) that has no way to be signaled. The Vulkan spec states: All elements of the pWaitSemaphores member of all elements of pSubmits must be semaphores that are signaled, or have semaphore signal operations previously submitted for execution (https://vulkan.lunarg.com/doc/view/1.2.170.0/linux/1.2-extensions/vkspec.html#VUID-vkQueueSubmit-pWaitSemaphores-00069)
    Objects: 2
        [0] 0x130000000013, type: 5, name: NULL
        [1] 0x3405438, type: 4, name: NULL
```

**Descriptor indexing** `vulkan_samples sample descriptor_indexing` - launch with no crash and no Validation errors.
On **resize crash** with same Validation error error.`vkQueueSubmit-pWaitSemaphores`

**Timeline semaphore** - on resize it crash/freeze, no Validation errors.

**Comment by gpx1000** [comment]
The startup validation errors on Buffer device address is because the glslang is generating the *EXT* instead of *KHR* SPIR-V.  This is fixed by updating the glslang submodule.
There's a glslang submodule update currently part of #356 Fixes this problem.
The resize issue is caused by prepared never being set to true in the prepare() function which in turn makes the resize function do nothing in the framework.  This is fixed in #382

**Comment by danilw** [comment]
Thanks for info, can be closed then.
